src/community/wps-jdbc/src/main/java/org/geoserver/wps/jdbc/JDBCStatusStore.java

```diff
@@ -0,0 +1,493 @@
+
+    static final String EXCEPTION_CLASS = "exceptionClass";
+
+    static final String ASYNC = "async";
+
+    static final String USER_NAME = "user";
+
+    static final String TASK = "task";
+
+    static final String SIMPLE_PROCESS_NAME = "processName";
+
+    static final String PROPERTIES = "properties";
+
+    static final String PROGRESS = "progress";
+
+    static final String PROCESS_NAME_URI = "processNameURI";
+
+    static final String PROCESS_NAME = "processNameImpl";
+
+    static final String PHASE = "phase";
+
+    static final String NODE_ID = "node";
+
+    static final String COMPLETION = "completionTime";
+
+    static final String LASTUPDATE = "lastUpdated";
+
+    static final String CREATION = "creationTime";
+
+    static final String PROCESS_ID = "processId";
+
+    static final String STATUS = "status";
+
+    static final String EXECUTION_ID = "exceptionId";
+
+    private static final String REQUEST = "request";
+
+    DataStore statuses;
+
+    static SimpleFeatureType schema;
+
+    public JDBCStatusStore(JDBCStatusStoreLoader loader) {
+        this(loader.getStore());
+    }
+
+    public JDBCStatusStore(DataStore store) {
+        if (store == null) {
+            throw new RuntimeException(
+                    "Attempted to create a JDBCStatusStore with a null datastore");
+        }
+        statuses = store;
+        try {
+            boolean statusSchema = statuses.getNames().contains(new NameImpl(STATUS));
+            if (statusSchema) {
+                LOGGER.info("using exisiting status store DB " + STATUS);
+                SimpleFeatureSource source = statuses.getFeatureSource(STATUS);
+                schema = source.getSchema();
+            } else {
+                LOGGER.info("creating new DB table for statuses");
+                try {
+                  //@formatter:off
+                    schema = DataUtilities.createType(STATUS,
+                              PROCESS_ID+":String," 
+                            + CREATION+":Date," 
+                            + LASTUPDATE+":Date,"
+                            + COMPLETION+":Date," 
+                            + NODE_ID+":String," 
+                            + PHASE+":String,"
+                            + PROCESS_NAME+":String," 
+                            + PROCESS_NAME_URI+":String,"
+                            + PROGRESS+":Float," 
+                            + REQUEST+":String," 
+                            + PROPERTIES+":String," 
+                            + SIMPLE_PROCESS_NAME+":String,"
+                            + TASK+":String," 
+                            + USER_NAME+":String," 
+                            + ASYNC+":Boolean,"
+                            + EXCEPTION_CLASS+":String,"
+                            + EXCEPTION_MESSAGE+":String," 
+                            + STACK_TRACE+":String");
+                } catch (SchemaException e) {
+                    // TODO Auto-generated catch block
```

Comment:
Ah, I see this one should actually never throw, sorry about that. But my comment is still valid for the catch below.